packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -1,169 +1,489 @@
+        onItemClick: (value: string) => {
+          if (readOnly) return;
+          if (disabled) return;
+          let item;
+          for (const valueItem of values) {
+            if (valueItem.value === value) {
+              item = valueItem;
+              break;
+            }
+          }
+          handleSelectItem(item as ValueItemType);
+        },
+      }}
+    >
+      <Box
+        className={classes.combobox}
+        ref={portalRef}
+      >
+        {/* This is only for the Combobox to work in forms */}
+        {name && <ComboboxNative />}
+
+        <ComboboxLabel />
+        <ComboboxInput />
+        {formFieldProps.hasError && <ComboboxError />}
+      </Box>
+
+      {/* This is the floating list with items */}
+      {open && (
+        <FloatingPortal root={portal ? null : portalRef}>
+          <FloatingFocusManager
+            context={context}
+            initialFocus={-1}
+            visuallyHiddenDismiss
+          >
+            <Box
+              shadow='medium'
+              borderRadius='medium'
+              aria-labelledby={formFieldProps.inputProps.id}
+              aria-autocomplete='list'
+              {...getFloatingProps({
+                ref: refs.setFloating,
+                style: {
+                  ...floatingStyles,
+                },
+              })}
+              className={cn(classes.itemsWrapper, classes[size])}
+            >
+              {/* Map our children, and add props if it is a ComboboxItem */}
+              {React.Children.map(filteredItems, (child, index) => {
+                if (
+                  React.isValidElement(child) &&
+                  child.type === ComboboxItem
+                ) {
+                  const props = {
+                    key: index,
+                    ref(node: HTMLElement | null) {
+                      listRef.current[index] = node;
+                    },
+                  };
+
+                  return React.cloneElement(child, {
+                    ...props,
+                  });
+                }
+                return child;
+              })}
+              {/* Add the rest of the children */}
+              {restChildren}
+            </Box>
+          </FloatingFocusManager>
+        </FloatingPortal>
+      )}
+    </ComboboxContext.Provider>
+  );
+};
+
+type ComboboxContextType = {
+  values: Set<ValueItemType>;
+  activeValues: ValueItemType[];
+  activeIndex: number | null;
+  multiple: boolean;
+  showEmptyChild: boolean;
+  disabled: boolean;
+  readOnly: boolean;
+  label: string | undefined;
+  description: string | undefined;
+  hideLabel: boolean;
+  open: boolean;
+  inputRef: React.RefObject<HTMLInputElement>;
+  refs: {
+    reference: React.MutableRefObject<ReferenceType | null>;
+    floating: React.MutableRefObject<HTMLElement | null>;
+    setReference: (node: ReferenceType | null) => void;
+    setFloating: (node: HTMLElement | null) => void;
```

Comment:
We should always use the types provided by the libraries we use instead of trying to infer them manually, this might get changed in the future and break stuff, which might make it hard to debug. Looks like the type for this is `UseFloatingReturn['refs']` (imported from `@floating-ui/react`)